repair-missing/script.whizzml

```diff
@@ -0,0 +1,192 @@
+;; This is a hack:  Field IDs are not conserved at batch prediction time, so
+;; we need to make sure that all of the fields have assigned IDs exactly as
+;; batch prediction will assign them (which essentially amounts to putting
+;; generated fields at the end).  To do this, we make a dummy model, do a dummy
+;; batch prediction, then remove the dummy column.
+(define (flatten-dataset dataset-id)
+  (let (fds ((fetch dataset-id) "fields"))
+    (if (some (lambda (fid) (= ((fds fid) "optype") "datetime")) (keys fds))
```

Comment:
I'm not sure if it counts as "generated", but my own troubles with the batchprediction name switcharoo are with an extended dataset that doesn't have a `datetime`. For example, take a dataset with fields 000000 through 00000e, and use a flatline expression to add a new field. By default, it will have id 10000f. Use that as the objective field of a logistic regression, and get the output dataset from a batchprediction, and now it's 00000f. 

But yes, jao seems to have fixed this! Wooo!